### PR TITLE
Added lesson details on page

### PR DESCRIPTION
**Pull request by Yurenko**
https://github.com/ita-social-projects/SpaceToStudy-Client/assets/83120263/5323df84-844a-40bb-a8a0-e300230706fd

### CODE REVIEW

File: src/pages/lesson-details/LessonDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  attachmentsMock,
+  defaultResponse
+} from '~/pages/lesson-details/LessonDetails.constants'
+import Accordions from '~/components/accordion/Accordions'
+import IconExtensionWithTitle from '~/components/icon-extension-with-title/IconExtensionWithTitle'
+
+import { errorRoutes } from '~/router/constants/errorRoutes'
+import { styles } from '~/pages/lesson-details/LessonsDetails.styles'
+import { Lesson, TypographyVariantEnum } from '~/types'
+import Loader from '~/components/loader/Loader'
```
Comment (Tolik170) [inline src/pages/lesson-details/LessonDetails.tsx:21]: move this import above

File: src/pages/lesson-details/LessonDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  const [activeItems, setActiveItems] = useState<number[]>([])
+
+  const { id } = useParams()
+  const navigate = useNavigate()
+
+  const onChange = (activeItem: number) => {
+    setActiveItems((prevActiveItems) => {
+      if (prevActiveItems.includes(activeItem)) {
+        return prevActiveItems.filter(
+          (prevActiveItem) => prevActiveItem !== activeItem
```
Comment (Tolik170) [inline src/pages/lesson-details/LessonDetails.tsx:33]: can you just return prevActiveItems or [...prevActiveItems] here?

File: src/pages/lesson-details/LessonDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    onResponseError: responseError
+  })
+
+  if (loading) {
+    return <Loader pageLoad />
+  }
+
+  if (!response) {
+    return null
+  }
```
Comment (Tolik170) [inline src/pages/lesson-details/LessonDetails.tsx:62]: do you need this if statement?

File: src/pages/lesson-details/LessonDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    <Box key={attachment.size} sx={styles.attachment}>
+      <IconExtensionWithTitle
+        size={attachment.size}
+        title={attachment.fileName}
+      />
+    </Box>
+  ))
+
+  const items = [
+    {
```
Comment (Tolik170) [inline src/pages/lesson-details/LessonDetails.tsx:74]: move this first item to constants

File: src/pages/lesson-details/LessonDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      title: 'lesson.attachments',
+      content: <Box sx={styles.attachmentList}>{attachmentsList}</Box>
+    }
+  ]
+
+  return (
+    <PageWrapper>
+      <Typography sx={styles.title}>{response.title}</Typography>
+      <Typography sx={styles.description}>{response.description}</Typography>
```
Comment (Tolik170) [inline src/pages/lesson-details/LessonDetails.tsx:88]: try to use TitleWithDescription component

File: src/pages/lesson-details/LessonDetails.tsx (hunk 1 above)
Comment (Yurenko) [reply src/pages/lesson-details/LessonDetails.tsx]: Fixed

File: src/pages/lesson-details/LessonDetails.tsx (hunk 3 above)
Comment (Yurenko) [reply src/pages/lesson-details/LessonDetails.tsx]: It was necessary when I didn't use navigate. But now I deleted it.

File: src/pages/lesson-details/LessonDetails.tsx (hunk 2 above)
Comment (Yurenko) [reply src/pages/lesson-details/LessonDetails.tsx:34]: If do it I will cannot close the accordion.

**Review by Tolik170 — APPROVED** [review]

**Review by Roman-Peretiatko — APPROVED** [review]

**Review by ArturBekhDEV — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**